File: 1921-eliminate-maximum-number-of-monsters/1921-eliminate-maximum-number-of-monsters.py
class Solution:
    def eliminateMaximum(self, dist: List[int], speed: List[int]) -> int:
        min_heap=[]
        for i in range(len(dist)):
            heappush(min_heap,(dist[i]+speed[i]-1)//speed[i])
        c=0
        while(min_heap):
            t=heappop(min_heap)
            if((t-c)<=0):
                return c
            c+=1
        return c

File: 1921-eliminate-maximum-number-of-monsters/1921-eliminate-maximum-number-of-monsters.java
class Solution {
    public int eliminateMaximum(int[] dist, int[] speed) {
        PriorityQueue<Integer> min_heap=new PriorityQueue<>();
        for(int i=0; i<dist.length; i++){
            min_heap.offer((dist[i]+speed[i]-1)/speed[i]);
        }
        int count=0;
        while(!min_heap.isEmpty()){
            int t=min_heap.poll();
            if((t-count)<=0) return count;
            count++;
        }
        return count;
    }
}